resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/IFile.java

```diff
@@ -414,7 +414,7 @@ public default void create(byte[] content, int createFlags, IProgressMonitor mon
 	 *                    in sync with the local file system
 	 * @param derived     Specifying this flag is equivalent to atomically calling
 	 *                    {@link IResource#setDerived(boolean)} immediately after
-	 *                    creating the resource or non-atomically setting the
+	 *                    creating the resource or atomically setting the
```

Comment:
Given the pain introduced by "atomically setting the derived flag before setting the content of an already" which as far as I understand is not a very interesting case (derived usually matters at creation), maybe we should just restrict it and make it "createAsDerived" and ignore it when resource already exists.

Reply:
The single most important usage of this API still is org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.writeClassFile(ClassFile, SourceFile, boolean) where the derived flag is updated also for existing files. Thats ancient code and i do not know why. But its legit. 
As a compromise we could only change (i.e. set) derived flag if derived==true is given as argument. In that case we would also not need the REPLACE. I pushed a implemementation for that: "A value of false will not update the derived flag of an existing file."